### PR TITLE
Fix code review findings

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary

- Fix `call_value` parameter description (contradicted TRX/SUN units)
- Use `strings.ReplaceAll` in `normalizeResult`
- `get_network` now returns active node address after failover via `pool.ActiveNode()`
- Instantiate keystore once per handler closure, not on every sign request
- Guard `SunToTRX` against `math.MinInt64` overflow
- Wrap `get_block` and `get_transaction` in `retry.Do` for consistency
- Upgrade health handler to `sync.RWMutex` with double-check pattern for cache hits
- Document `AddFallback` must be called before concurrent use
- Validate TRC20 decimals range (0-77) in `get_trc20_balance` and `transfer_trc20`
- Add 1MB max length validation on hex string inputs (sign, broadcast, decode_abi_output)

## Test plan

- [x] `make fmt` — no changes
- [x] `make lint` — 0 issues
- [x] `make test` — all pass

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed SUN→TRX edge-case formatting for minimum int64.
  * Validated TRC20 decimals (0–77) to prevent invalid operations.
  * Normalized network results ("SUCESS" → "SUCCESS").

* **Improvements**
  * Health endpoint serves cached responses without blocking writers and refreshes safely when expired.
  * Added retry logic for block/transaction RPCs to improve reliability.
  * Enforced a 1 MB limit on transaction/hex inputs and hardened large-response handling.
  * Network calls now report the active node used and fetch latest block reliably.

* **Tests**
  * Added test for SUN→TRX edge-case.

### CODE REVIEW

File: internal/tools/network.go [hunk 1]
```diff
@@ -58,12 +60,16 @@ func handleGetTransaction(pool *nodepool.Pool) server.ToolHandlerFunc {
...
+			return grpc.GetTransactionByID(txID)
+		})
 		if err != nil {
 			return mcp.NewToolResultError(fmt.Sprintf("get_transaction: %v", err)), nil
 		}
 
-		info, err := grpc.GetTransactionInfoByID(txID)
+		info, err := retry.Do(func() (*core.TransactionInfo, error) {
+			return grpc.GetTransactionInfoByID(txID)
+		})
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/network.go]: Fixed — same pattern applied. Both retry closures now call `pool.Client()` inside the callback.

File: internal/tools/token.go [hunk 2]
```diff
@@ -72,14 +72,19 @@ func handleGetTRC20Balance(pool *nodepool.Pool) server.ToolHandlerFunc {
 			symbol = ""
 		}
 
+		if decimals == nil || decimals.Sign() < 0 || decimals.Cmp(big.NewInt(77)) > 0 {
+			return mcp.NewToolResultError(fmt.Sprintf("get_trc20_balance: invalid decimals value: %s", decimals)), nil
+		}
+		dec := int(decimals.Int64())
+
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/token.go]: Fixed — added the same `nil/Sign()/Cmp(big.NewInt(77))` validation before `Int64()` in `handleGetTRC20TokenInfo`, matching the pattern in `handleGetTRC20Balance`.
